tests/integration/states/test_ansiblegate.py

```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8 -*-
+'''
+Test AnsibleGate State Module
+'''
+from __future__ import absolute_import, unicode_literals, print_function
+
+# Import python libraries
+import os
+import shutil
+import tempfile
+import yaml
+
+# Import salt libraries
+import salt.utils.files
+import salt.utils.path
+
+# Import testing libraries
+from tests.support.case import ModuleCase
+from tests.support.helpers import destructiveTest
+from tests.support.mixins import SaltReturnAssertsMixin
+from tests.support.runtests import RUNTIME_VARS
+from tests.support.unit import skipIf
+
+
+@destructiveTest
+@skipIf(not salt.utils.path.which('ansible-playbook'), 'ansible-playbook is not installed')
+class AnsiblePlaybooksTestCase(ModuleCase, SaltReturnAssertsMixin):
+    '''
+    Test ansible.playbooks states
+    '''
+    def setUp(self):
+        priv_file = os.path.join(RUNTIME_VARS.TMP_CONF_DIR, 'key_test')
+        data = {
+            'all': {
+                'hosts': {
+                    'localhost': {
+                        'ansible_host': '127.0.0.1',
+                        'ansible_port': 2827,
+                        'ansible_user': RUNTIME_VARS.RUNNING_TESTS_USER,
+                        'ansible_ssh_private_key_file': priv_file,
+                        'ansible_ssh_extra_args': (
+                            '-o StrictHostKeyChecking=false '
+                            '-o UserKnownHostsFile=/dev/null '
+                        )
+                    },
+                },
+            },
+        }
+        self.tempdir = tempfile.mkdtemp()
+        self.inventory = self.tempdir + 'inventory'
+        with salt.utils.files.fopen(self.inventory, 'w') as yaml_file:
+            yaml.dump(data, yaml_file, default_flow_style=False)
+
+    def tearDown(self):
+        shutil.rmtree(self.tempdir)
+        delattr(self, 'tempdir')
+        delattr(self, 'inventory')
+
+    def test_ansible_playbook(self):
+        ret = self.run_state(
+            'ansible.playbooks',
+            name='remove.yml',
+            gitrepo='git://github.com/gtmanfred/playbooks.git',
+            ansible_kwargs={'inventory': self.inventory}
+        )
+        self.assertSaltTrueReturn(ret)
+        ret = self.run_state(
+            'ansible.playbooks',
+            name='install.yml',
+            gitrepo='git://github.com/gtmanfred/playbooks.git',
```

Comment:
There is no such param as `gitrepo` anymore (and so this should be fixed). But why the test is passing anyway, makes me wondering if the `run_state` is reliable enough reporting wrong parameter has been passed?

Reply:
It did fail this time https://jenkins.saltstack.com/job/PR/job/salt-pr-linode-ubuntu14-n/23939/testReport/junit/integration.states.test_ansiblegate/AnsiblePlaybooksTestCase/test_ansible_playbook/

With not having gitrepo in the keys.